frontend/src/metabase/App.tsx

```diff
@@ -93,25 +98,43 @@ function App({
   children,
 }: AppProps) {
   const [viewportElement, setViewportElement] = useState<HTMLElement | null>();
-  const [errorInfo, setErrorInfo] = useState<ErrorInfo | null>(null);
+  const [errorInfo, setErrorInfo] = useState<(ErrorInfo & Error) | null>(null);
+  const [errorCount, setErrorCount] = useState(0);
+
+  const countError = () => setErrorCount(prev => prev + 1);
+  console.log("errorInfo", errorInfo);
 
   useOnMount(() => {
     initializeIframeResizer();
   });
 
   return (
-    <ErrorBoundary onError={setErrorInfo}>
+    <ErrorBoundary onError={setErrorInfo} countError={countError}>
       <ScrollToTop>
         <AppContainer className="spread">
           <AppBanner />
           {isAppBarVisible && <AppBar isNavBarVisible={isNavBarVisible} />}
           <AppContentContainer isAdminApp={isAdminApp}>
             {isNavBarVisible && <Navbar />}
-            <AppContent ref={setViewportElement}>
-              <ContentViewportContext.Provider value={viewportElement ?? null}>
-                {errorPage ? getErrorComponent(errorPage) : children}
-              </ContentViewportContext.Provider>
-            </AppContent>
+            {errorCount < MAX_ERRORS_ALLOWED ? (
+              <AppContent ref={setViewportElement}>
+                <ContentViewportContext.Provider
+                  value={viewportElement ?? null}
+                >
+                  {errorPage ? getErrorComponent(errorPage) : children}
+                </ContentViewportContext.Provider>
+              </AppContent>
+            ) : (
+              getErrorComponent({
+                status: 500,
+                data: {
+                  error_code: "looping error",
+                  message:
+                    (errorInfo?.message ?? "") +
+                    (errorInfo?.componentStack ?? ""),
```

Comment:
Would there be a case where we have both `message` and `componentStack` and there won't be a space to separate between the 2 strings?